### PR TITLE
Fix an issue when using gtest with Conan 2

### LINKED ISSUE #12884: [package] gtest/1.12.1: Conan 2 compatibility
### Package and Environment Details

* Package Name/Version: **gtest/1.12.1**
* Operating System+version: **Windows 10**
* Compiler+version: **MSVC 19.33.31629**
* Conan version: **conan 2.0.0-beta2**
* Python version: **Python 3.10.7**

### Conan profile

```
> conan profile show
Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows

Build profile:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

### Steps to reproduce

Build a library which depends on gtest.

### Logs

```
ERROR: gtest/1.12.1: Error in validate() method, line 89
        if self.options.shared and (is_msvc(self) or self._is_clang_cl) and "MT" in msvc_runtime_flag(self):
        ConanException: 'self.options' access in 'validate()' method is forbidden
```

### PR DESCRIPTION
**Pull request by mark0n**
**gtest/1.12.1**

This fixes #12884.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gtest/all/conanfile.py [hunk 1]
```diff
@@ -86,13 +86,16 @@ def package_id(self):
         del self.info.options.no_main
 
     def validate(self):
-        if self.options.shared and (is_msvc(self) or self._is_clang_cl) and "MT" in msvc_runtime_flag(self):
+        shared = self.info.options.shared
+        if (shared and (is_msvc(self) or self._is_clang_cl)
+                and "MT" in msvc_runtime_flag(self)):
```
Comment (prince-chrismc) [inline recipes/gtest/all/conanfile.py:91]: There's a new helper for this https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc-static-runtime

Should clean up + prevent the issue

File: recipes/gtest/all/conanfile.py [hunk 2]
```diff
@@ -101,10 +104,10 @@ def loose_lt_semver(v1, v2):
             min_length = min(len(lv1), len(lv2))
             return lv1[:min_length] < lv2[:min_length]
 
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if min_version and loose_lt_semver(str(self.settings.compiler.version), min_version):
+        min_version = self._minimum_compilers_version.get(str(compiler))
+        if min_version and loose_lt_semver(str(compiler.version), min_version):
             raise ConanInvalidConfiguration(
-                f"{self.name} requires {self.settings.compiler} {min_version}. The current compiler is {self.settings.compiler} {self.settings.compiler.version}."
+                f"{self.name} requires {compiler} {min_version}. The current compiler is {compiler} {compiler.version}."
```
Comment (prince-chrismc) [inline recipes/gtest/all/conanfile.py:110]: just a complete name

```suggestion
                f"{self.ref} requires {compiler} {min_version}. The current compiler is {compiler} {compiler.version}."
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution!

File: recipes/gtest/all/conanfile.py [hunk 3]
```diff
@@ -86,13 +86,16 @@ def package_id(self):
...
+        if (shared and (is_msvc(self) or self._is_clang_cl)
+                and "MT" in msvc_runtime_flag(self)):
             raise ConanInvalidConfiguration(
                 "gtest:shared=True with compiler=\"Visual Studio\" is not "
                 "compatible with compiler.runtime=MT/MTd"
             )
 
-        if self.settings.compiler.get_safe("cppstd"):
+        compiler = self.info.settings.compiler
+        if compiler.get_safe("cppstd"):
```
[the remark this answers is not in the document]
Comment (mark0n) [reply recipes/gtest/all/conanfile.py]: I included the change you proposed.

File: recipes/gtest/all/conanfile.py [hunk 4]
```diff
@@ -86,13 +86,13 @@ def package_id(self):
         del self.info.options.no_main
 
     def validate(self):
-        if self.options.shared and (is_msvc(self) or self._is_clang_cl) and "MT" in msvc_runtime_flag(self):
+        if is_msvc_static_runtime(self) and self.info.options.shared(self):
```
Comment (uilianries) [inline recipes/gtest/all/conanfile.py:89]:
```suggestion
        if is_msvc_static_runtime(self) and self.info.options.shared:
```

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Review by prince-chrismc — APPROVED** [review]
There's a small linter comment

**Review by danimtb — APPROVED** [review]

**Pull request merged**